pkg/checkup/workspace.go

```diff
@@ -0,0 +1,346 @@
+package checkup
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/client-go/kubernetes"
+	"log"
+	"strings"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	checkupNamespaceName           = "checkup-workspace"
+	resultsConfigMapName           = "checkup-results"
+	checkupJobName                 = "checkup-job"
+	serviceAccountName             = "checkup-sa"
+	resultsConfigMapWriterRoleName = "results-configmap-writer"
+)
+
+const (
+	resultsConfigMapNameEnvVarName      = "RESULT_CONFIGMAP_NAME"
+	resultsConfigMapNameEnvVarNamespace = "RESULT_CONFIGMAP_NAMESPACE"
+)
+
+const (
+	ClusterRoleKind = "ClusterRole"
+	RoleKind        = "Role"
+)
+
+type workspace struct {
```

Comment:
Consider adding the spec here

Reply:
There is no actual need for the `spec` object it is being used only on `NewCheckupWorkspace` as an input.
This is why I didnt added it in the first place.